data_fetcher.py

```diff
@@ -0,0 +1,118 @@
+
+""" Scrapes the web and caches data to disk for the Biased News Sentiment program.
+
+    Written by Kyle Combes for Software Design (Spring 2017) Mini Project 3
+    at Olin College of Engineering.
+"""
+import os.path
+import requests
+import untangle
+import json
+from bs4 import BeautifulSoup
+from data_structures import *
+
+class DataFetcher:
+
+    headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; WOW64; rv:51.0) Gecko/20100101 Firefox/51.0'}
+    FILE_PREFIX = 'data/'
```

Comment:
Readers will probably not know what this piece of code does. For non-trivial piece of code like this, an inline-comment describing what each part of the code does is a good idea.